contracts/utils/Panic.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+/**
+ * @dev Helper library for emitting standardized panic codes.
+ */
+// slither-disable-next-line unused-state
+library Panic {
+    uint256 internal constant ASSERTION_ERROR = 0x1;
+    uint256 internal constant ARITHMETIC_UNDER_OR_OVERFLOW = 0x11;
+    uint256 internal constant DIVISION_BY_ZERO = 0x12;
+    uint256 internal constant ENUM_CONVERSION_OUT_OF_BOUNDS = 0x21;
+    uint256 internal constant INCORRECTLY_ENCODED_STORAGE_BYTE_ARRAY = 0x22;
+    uint256 internal constant POP_ON_EMPTY_ARRAY = 0x31;
+    uint256 internal constant ARRAY_ACCESS_OUT_OF_BOUNDS = 0x32;
+    uint256 internal constant TOO_MUCH_MEMORY_ALLOCATED = 0x41;
+    uint256 internal constant ZERO_INITIALIZED_VARIABLE = 0x51;
+
+    function panic(uint256 code) internal pure {
```

Comment:
I'm not too worried about users emitting random panic code ... but I'm not fundamentally opposed to making `panic(uint256)` private and having functions like
```
function TooMuchMemoryAllocated() internal pure { _panic(0x41); }
```

We would have to make them CamelCase though.

Reply:
The thing, if new panic code are ever added, the current version would allow users to use them without us having to do a release. 
I'm also thinking, what if some L2 whant to use specific panic codes for some crosschain stuff. We won't want to add them to the library, but users will want to emit them.

Reply:
The linter rules are not ok with that, and I think we should follow them:
- constant must be SNAKE_CASE
- internal function in libraries are not _prefixed.

> The developers will have defined functions out of the box without the need to check the internal constants.

I'm not sure that is a good point. When testing panic code with hardhat you do 
```
expect(...).to.be.revertedWithPanic(PANIC_CODE.ARITHMETIC_UNDER_OR_OVERFLOW)
```
you don't to
```
expect(...).to.be.revertedWithPanicArithmeticUnderOrOverflow();
```

Reply:
I think the cureent version is simple and effective. It's targetting advanced users anyway.

Reply:
> I'm not sure that is a good point. When testing panic code with hardhat you do
> 
> ```
> expect(...).to.be.revertedWithPanic(PANIC_CODE.ARITHMETIC_UNDER_OR_OVERFLOW)
> ```

Fair

> I think the cureent version is simple and effective. It's targetting advanced users anyway.

We should document the function then. Currently it allows panicking with anything it I think it needs some usage instructions in both the contracts NatSpec and the `panic` funciton itself